src/main/java/org/openmrs/addonindex/backend/BackendHandler.java

```diff
@@ -30,5 +30,6 @@
 	 * @throws Exception
 	 */
 	AddOnInfoAndVersions getInfoAndVersionsFor(AddOnToIndex addOnToIndex) throws Exception;
+	void fetchDownloadCounts(AddOnToIndex toIndex, AddOnInfoAndVersions infoAndVersions) throws Exception;
```

Comment:
Needs javadoc to explain what an implementation of this interface should do.

Reply:
Also, now that I'm realizing that the other tools don't handle this, I suggest doing it in a slightly different way.
Introduce new subinterface like

    public interface SupportsDownloadCounts extends BackendHandler

Then you only implement this for Bintray and don't have to have an empty method in the other classes.